.github/workflows/android.yml

```diff
@@ -75,19 +75,45 @@ jobs:
         with:
           path: example/android/app/build
           key: ${{ runner.os }}-build-${{ env.cache-name }}-example-${{ hashFiles('**/react-native/package.json') }}
-  
-      - name: SKDs - download required images
-        run: $ANDROID_HOME/cmdline-tools/latest/bin/sdkmanager "system-images;android-30;default;x86_64"
 
-      - name: Emulator - Create
-        run: $ANDROID_HOME/cmdline-tools/latest/bin/avdmanager create avd -n Pixel_API_30 --device 'Nexus 5X' --package "system-images;android-30;default;x86_64" --sdcard 512M
+      - name: Build Android app
+        working-directory: example/android
+        run: ./gradlew assembleDebug
 
-      - name: Emulator - Boot
-        run: $ANDROID_HOME/emulator/emulator -memory 4096 -avd Pixel_API_30 -wipe-data -no-window -gpu swiftshader_indirect -no-snapshot -noaudio -no-boot-anim &
+      - uses: actions/upload-artifact@v2
+        with:
+          path: example/android/app/build/outputs/apk/debug/app-debug.apk
+          name: app-debug.apk
+  test:
+    needs: build
+    runs-on: macos-12
+    concurrency:
+      group: android-example-${{ github.ref }}
+      cancel-in-progress: true
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
 
-      - name: ADB Wait For Device
-        run: adb wait-for-any-device
-        timeout-minutes: 3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+
+      - name: Install root node dependencies
+        run: yarn
+      
+      - name: Download APK
+        uses: actions/download-artifact@v3
+        with:
+          name: app-debug.apk
+      
+      - name: Start Emulator
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 29
+          script: ./gradlew connectedCheck
```

Comment:
This will still do a full build before deploying the APK and running tests. But I'll leave it up to the team to figure out what you want :)